### PR TITLE
Add eslint and code format configuration

### PR DESCRIPTION
**Pull request by danielkhan**
Added eslint and copied all config files for code style over from fastify to make contributing easier.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — COMMENTED** [review]
If we install this packages, I think we should at least use it in command instead of IDE only.

The command for linting.
https://github.com/fastify/fastify/blob/main/package.json#L13-L16

**Comment by danielkhan** [comment]
> If we install this packages, I think we should at least use it in command instead of IDE only.

I tried but looking closer, it seems as if the `eslint` command is only used to validate typescript while eslint by configuration extends standard. By running `standard` on `npm test` the same rules the IDE uses through eslint are applied.

File: package.json [hunk 1]
```diff
@@ -39,11 +39,20 @@
...
     "cross-env": "^7.0.2",
     "dot": "^1.1.3",
     "ejs": "^3.1.2",
+    "eslint": "^7.32.0",
+    "eslint-config-standard": "^16.0.3",
+    "eslint-import-resolver-node": "^0.3.6",
+    "eslint-plugin-import": "^2.25.3",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-promise": "^5.1.1",
+    "eslint-plugin-standard": "^5.0.0",
```
Comment (Eomm) [inline package.json:55]: installing `standard` only, all these dep are not necessary (it ease the dependabot PRs etc..)

Comment (danielkhan) [reply package.json:55]: You lose all the IDE help though, don't you?
Also, this is 1:1 what fastify installs.
Happy to remove it but I wonder why this shouldn't be consistent with fastify.

Comment (Eomm) [reply package.json:55]: > You lose all the IDE help though, don't you?

It did not happen to me 😕 

> Happy to remove it but I wonder why this shouldn't be consistent with fastify.

I must say that I was not aware of those deps 😱 
I would like to check, but it is fine - it is just a matter of preferences

**Review by Eomm — APPROVED** [review]

**Comment by danielkhan** [comment]
@climba03003 are we good to go? Don't want to gloss over your objection.

**Review by climba03003 — COMMENTED** [review]
My stand is the same as before, but I am not blocking it to be merged.

From my experience, the first thing after cloning the project always be `disable IDE auto linting`. Because I can't get it works even once, `fastify` is the same in my environment.

So, I only use the `command` before merging but not rely on the auto linting feature of IDE. And that is the reason why I think if you add it, you need to use it. As it do nothing in my environment, just increase the installation time.

**Comment by danielkhan** [comment]
> So, I only use the command before merging but not rely on the auto linting feature of IDE. And that is the reason why I think if you add it, you need to use it. As it do nothing in my environment, just increase the installation time.

Understood.
Technically it is used as most IDEs will look inside the current workspace when trying to find eslint.
But the fact that it does now work on my machine, is of course not too meaningful. Yet sticking to the same ruleset as fastify makes sense in my opinion.

**Review by Eomm — CHANGES_REQUESTED** [review]
I would like to ask you for one more test on your PC:
starting from `master`, could you add a simple `.eslintrc` like this: https://github.com/standard/vscode-standard/issues/295#issuecomment-962632894

cleaning your `node_modules` and without the additions?

I think you hit the same issue I'm having and the `.eslintrc` should be enough

For example, using this PR:

```
$ npm ls eslint-config-standard
point-of-view@4.15.3 /Users/mspigolon/workspace/point-of-view
├── eslint-config-standard@16.0.3
└─┬ standard@16.0.4
  └── eslint-config-standard@16.0.3 deduped
```

**Comment by danielkhan** [comment]
> I would like to ask you for one more test on your PC: starting from `master`, could you add a simple `.eslintrc` like this: 

I've done that, cleaned node_modules, reinstalled without these deps.
Turns out that Standard itself has the dependencies already, so I omitted them in my package.json now.

```
dkhan@mba-prv point-of-view % npm ls eslint
point-of-view@4.15.3 /Users/dkhan/CodeKitchen/point-of-view
└─┬ standard@16.0.4
  ├─┬ eslint-config-standard-jsx@10.0.0
  │ └── eslint@7.18.0 deduped
  ├─┬ eslint-config-standard@16.0.3
  │ └── eslint@7.18.0 deduped
  ├─┬ eslint-plugin-import@2.24.2
  │ └── eslint@7.18.0 deduped
  ├─┬ eslint-plugin-node@11.1.0
  │ ├─┬ eslint-plugin-es@3.0.1
  │ │ └── eslint@7.18.0 deduped
  │ └── eslint@7.18.0 deduped
  ├─┬ eslint-plugin-promise@5.1.1
  │ └── eslint@7.18.0 deduped
  ├─┬ eslint-plugin-react@7.25.3
  │ └── eslint@7.18.0 deduped
  └── eslint@7.18.0
```

**Review by climba03003 — APPROVED** [review]
LGTM.

**Review by Eomm — APPROVED** [review]

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ Set the default behavior, in case people don't have core.autocrlf set
```
Comment (Eomm) [inline .gitattributes:1]:
```suggestion
# Set the default behavior, in case people don't have core.autocrlf set
```

**Comment by mcollina** [comment]
I think I landed the other PR first :( Can you rebase this?

**Comment by danielkhan** [comment]
> I think I landed the other PR first :( Can you rebase this?

I was able to merge this.

**Comment by climba03003** [comment]
> > I think I landed the other PR first :( Can you rebase this?
> 
> I was able to merge this.

Can you remove the `eslint` related plugin inside this PR. As they are added in another PR.

**Comment by danielkhan** [comment]
> Can you remove the `eslint` related plugin inside this PR. As they are added in another PR.

Correct. Done.

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**